### ISSUE TITLE
Add --polycos option for phase calculation

### ISSUE THREAD

**Issue by sguillot** [paulray/NICERsoft#80]
The new option `--polycos` in the PINT photonphase script significantly improves the calculation time for large event files (https://github.com/nanograv/PINT/pull/1199).

We should consider adding this option to `psrpipe.py` and `merge.py`.

**Comment by paulray** [comment]
Indeed! As soon as that is tested and merged, we should take advantage of it!

**Comment by paulray** [comment]
However, for NICER data there is a significant optimization needed. It shouldn't be hard but Sasha doesn't have time to work on it.
https://github.com/nanograv/PINT/issues/1209

**Comment by sguillot** [comment]
@paulray, what is the status of the use of `--polycos` in PINT ?

**Comment by paulray** [comment]
It should be working, but it requires barycentered photons.
Please try it out!

**Comment by sguillot** [comment]
This could be implemented at the same time as #28, i.e., psrpipe could call `barycorr` if needed.

The question is which coordinates to use for the barycentric correction:

1. from the event file header ? 
2. user-provided coordinates ?
3.  the PAR file ?

Thoughts?

**Comment by paulray** [comment]
Yes it could barycenter the data. That would create a new evt file, which would have the phases, but actually the PULSE_PHASE column could even be copied back to the topocentric event file. The barycentering should use the .par file coordinates, but there are options here: (1) use RAJ,DECJ or (2) use RAJ,DECJ accounting for PMRA and PMDEC or (3) use something other than barycorr to do the barycentering.  It will be important to match the ephemeris used for barycorr with the one specified in the par file.
